packages/@react-spectrum/tabs/src/Tabs.tsx

```diff
@@ -34,64 +37,124 @@ export function Tabs<T extends object>(props: SpectrumTabsProps<T>) {
     density,
     ...otherProps
   } = props;
+
   let ref = useRef<HTMLDivElement>();
+  let wrapperRef = useRef<HTMLDivElement>();
   let state = useSingleSelectListState<T>({
     ...props,
     onSelectionChange
   });
 
+  let {direction} = useLocale();
   let {styleProps} = useStyleProps(otherProps);
   let {tabListProps, tabPanelProps} = useTabs(props, state, ref);
+  let [collapse, setCollapse] = useValueEffect(false);
   let [selectedTab, setSelectedTab] = useState<HTMLElement>();
 
   useEffect(() => {
-    let tabs: HTMLElement[] = Array.from(ref.current.querySelectorAll('.' + styles['spectrum-Tabs-item']));
-    setSelectedTab(tabs.find((tab) => tab.dataset.key === state.selectedKey));
-  }, [props.children, state.selectedKey]);
+    // Ensure a tab is always selected (in case no selected key was specified or if selected item was deleted from collection)
+    if (state.selectionManager.isEmpty || !state.collection.getItem(state.selectedKey)) {
+      state.selectionManager.replaceSelection(state.collection.getFirstKey());
+    }
+  }, [state]);
```

Comment:
Do we want to listen to the entire state or just the selectedKey or selectionManager?